src/nccl_ofi_net.c

```diff
@@ -1050,49 +1064,130 @@ static inline ncclResult_t nccl_ofi_progress(nccl_ofi_t *nccl_ofi_comp)
 }
 
 /*
- * @brief	Checks if the underlying hardware type matches the input platform type.
+ * @brief	Provides EC2 platform type as reported by the
+ * 		first line of
+ *		/sys/devices/virtual/dmi/id/product_name.
+ *		Users of this API *should* free the buffer when a
+ *		Non-NULL string is returned.
  *
- * @return	-1, on error
- *		1, if it matches
- *		0, if it does not match
+ * @return	NULL, on allocation and file system error
+ * 		EC2 platform type, on success
  */
-static int platform_type_matches(char *platform_type)
+static char* get_platform_type()
```

Comment:
I still think this is overkill (will we ever actually have a model name over say 256 chars?) but I'll disagree and commit.

Reply:
This is just me being pedantic and holding the file as the source of truth :)